pkg/cluster/spec/spec.go

```diff
@@ -495,9 +501,30 @@ func (s *Specification) CountDir(targetHost, dirPrefix string) int {
 	return count
 }
 
+func (s *Specification) validateTiSparkSpec() error {
+	// There must be a Spark master
+	if len(s.TiSparkMasters) == 0 {
+		if len(s.TiSparkWorkers) == 0 {
+			return nil
+		}
+		return ErrNoTiSparkMaster
+	}
+
+	// We only support 1 Spark master at present
+	if len(s.TiSparkMasters) > 1 {
+		return ErrMultipleTiSparkMaster
+	}
+
+	return nil
+}
+
 // Validate validates the topology specification and produce error if the
 // specification invalid (e.g: port conflicts or directory conflicts)
 func (s *Specification) Validate() error {
+	if err := s.validateTiSparkSpec(); err != nil {
```

Comment:
I think this should be checked after other checks (eg. platformConflictsDetect)
For example, I have this scale-out.yaml:
```
tidb_servers:
  - host: 172.16.5.139
  - host: 172.16.5.139
```
This is clearly in conflict. However, if there is a `ErrNoTiSparkMaster` error, it may be ignored in https://github.com/pingcap/tiup/pull/531/files#diff-19d3b6c1bf7701c04716d697df146cc1R100